scipy/special/xsf/digamma.h

```diff
@@ -168,7 +168,7 @@ XSF_HOST_DEVICE inline std::complex<double> digamma(std::complex<double> z) {
          *
          *https://dlmf.nist.gov/5.5#E4
          */
-        res = -M_PI * cospi(z) / sinpi(z);
+        res = -M_PI * cotpi(z);
```

Comment:
Will this cause different behaviour for edge cases in `digamma`? Might be overthinking but if yes, it would be  a backwards compatability break.

Reply:
By edge cases, do you mean the poles at non-positive integers? They are already handled in an earlier branch.

As an aside, it's long overdue for us to clean up the behavior of `gamma` and friends at poles. We systematically get these wrong. See for instance, https://github.com/scipy/scipy/issues/19071. I'm still unsure whether this will require going through a deprecation process since it is technically a backwards compatibility break, though it could also be considered a bug. I guess you would lean on the side of this needing a deprecation process.

Reply:
If the current behaviour is definitely wrong by standard definition of these functions, then I am fine without a deprecation cycle.